systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java

```diff
@@ -912,6 +881,101 @@ void testClusterCaRemovedTriggersRollingUpdate() {
+        kafkaRule.put("pattern", "kafka.(\\w+)<type=(.+), name=(.+)><>Count");
+        kafkaRule.put("name", "kafka_$1_$2_$3_count");
+        kafkaRule.put("type", "COUNTER");
+
+        Map<String, Object> kafkaMetrics = new HashMap<>();
+        kafkaMetrics.put("lowercaseOutputName", true);
+        kafkaMetrics.put("rules", Collections.singletonList(kafkaRule));
+
+        //Zookeeper
+        Map<String, Object> zookeeperLabels = new HashMap<>();
+        zookeeperLabels.put("replicaId", "$2");
+
+        Map<String, Object> zookeeperRule = new HashMap<>();
+        zookeeperRule.put("labels", zookeeperLabels);
+        zookeeperRule.put("name", "zookeeper_$3");
+        zookeeperRule.put("pattern", "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+)><>(\\w+)");
+
+        Map<String, Object> zookeeperMetrics = new HashMap<>();
+        zookeeperMetrics.put("lowercaseOutputName", true);
+        zookeeperMetrics.put("rules", Collections.singletonList(zookeeperRule));
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 3)
+                .editSpec()
+                    .editKafka()
+                        .withMetrics(kafkaMetrics)
+                    .endKafka()
+                    .editOrNewZookeeper()
+                        .withMetrics(zookeeperMetrics)
+                    .endZookeeper()
+                    .withNewKafkaExporter()
+                    .endKafkaExporter()
+                .endSpec()
+                .done();
+
+        LOGGER.info("Check if metrics are present in pod of Kafka and Zookeeper");
+        String kafkaMetricsOutput = MetricsUtils.collectMetrics(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "");
+        String zkMetricsOutput = MetricsUtils.collectMetrics(CLUSTER_NAME + "-zookeeper-0", "");
+        assertThat(kafkaMetricsOutput, containsString("kafka_"));
+        assertThat(zkMetricsOutput, containsString("replicaId"));
+
+        LOGGER.info("Changing metrics to something else");
+        kafkaRule.replace("pattern", "kafka.(\\w+)<type=(.+), name=(.+)><>Count",
+                "kafka.(\\w+)<type=(.+), name=(.+)Percent\\w*><>MeanRate");
+        kafkaRule.replace("name", "kafka_$1_$2_$3_count", "kafka_$1_$2_$3_percent");
+        kafkaRule.replace("type", "COUNTER", "GAUGE");
+
+        zookeeperRule.replace("pattern",
+                "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+)><>(\\w+)",
+                "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+)><>(\\w+)");
+        zookeeperRule.replace("name", "zookeeper_$3", "zookeeper_$2");
+        zookeeperRule.replace("labels", zookeeperLabels, null);
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            kafka.getSpec().getKafka().setMetrics(kafkaMetrics);
+            kafka.getSpec().getZookeeper().setMetrics(zookeeperMetrics);
+        });
+
+        LOGGER.info("Check if Kafka and Zookeeper metrics are changed");
```

Comment:
I think the original test checked that no rolling update happened. But the new test seems to be missing it. Could you add it?

Reply:
Yes I missed it, thanks